### ISSUE TITLE
Support conditional composite receipts in proof composition

### ISSUE THREAD

**Issue by nategraf** [risc0/risc0#982]
With proof composition, there is the concept of a "conditional receipt". A conditional receipt is a receipt that is valid only if any assumptions made by the guest during execution can be proven by other corroborating receipts.

This issue is to implement conditional receipts through the prover and `CompositeReceipt` struct.

Related: #980, #981

**Comment by flaub** [comment]
@nategraf Did this make it into 1.0?

**Comment by nategraf** [comment]
No, because it requires the support for unresolved assumptions. The API is designed to support it, but the actual support is not implemented.

**Comment by saefstroem** [comment]
Any update on this?

**Comment by nategraf** [comment]
@saefstroem, just to bring the context from Discord to here: we scoped this feature, but so far have not found a use case that really needs it. Because we are always short on time, this is one that has never reached the top of priority list.

**Comment by saefstroem** [comment]
> @saefstroem, just to bring the context from Discord to here: we scoped this feature, but so far have not found a use case that really needs it. Because we are always short on time, this is one that has never reached the top of priority list.

Right makes sense and is understandable. We have found other ways to go about our problem so it is fine. This is no longer a priority for me either.
